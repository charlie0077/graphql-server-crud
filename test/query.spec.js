import ApolloClient, { gql } from 'apollo-boost'
import 'cross-fetch/polyfill'

export const client = new ApolloClient({
  uri: 'http://localhost:4000/',
  onError: (e) => { console.log(e) }
})

test('test simple query', async () => {
  const query = gql`
    query {
      getAuthor(id: "1") {
        id
        email
      }
    }
  `

  expect(await client.query({ query: query })).toMatchSnapshot()
})

test('test nested query', async () => {
  const query = gql`
    query {
      queryPost(
        where: { id: { gt: "1" } }
        offset: 0
        limit: 3
        orderBy: { column: "id", order: "asc" }
      ) {
        id
        title
        author {
          email
          company {
            domain
          }
        }
      }
    }
  `

  expect(await client.query({ query: query })).toMatchSnapshot()
})

test('test aggregation', async () => {
  const query = gql`
    query {
      queryAuthorJoinPost(
        where: {
          post_id: { gt: "2" }
        }
        orderBy: [{ column: "id", order: "asc" }]
        limit: 10
        offset: 0
        groupBy: ["id", "email"]
        having: {
          age__avg: {gt: 30}
        }
      ) {
        id
        email
        age__avg
      }
    }
  `

  expect(await client.query({ query: query })).toMatchSnapshot()
})

test('test complicate search', async () => {
  const query = gql`
    query {
      queryAuthor(
        where: {
          _or: [
            {
              email: { in: ["email1"] }
              last_name: { in: ["name1"] }
              _or: [{ id: { eq: "1" } , last_name: { in: ["name2"] }}]
            }
            { first_name: { in: ["xx", "xx"] } }
          ],
          _and: [
            {
              email: { in: ["email2"] }
              last_name: { in: ["name3"] }
              _or: [{ id: { eq: "2" } , last_name: { in: ["name4"] }}]
            }
            { first_name: { in: ["name5", "name6"] } }
          ]
        }
      ) {
        id
        email
        first_name
        last_name
      }
    }
  `

  expect(await client.query({ query: query })).toMatchSnapshot()
})

test('test limit and offset', async () => {
  const query = gql`
    query {
      queryPost(
        offset: 0
        limit: 3
      ) {
        id
        title
      }
    }
  `

  expect(await client.query({ query: query })).toMatchSnapshot()
})

test('test sort', async () => {
  const query = gql`
    query {
      queryPost(
        where: { id: { gt: "1" } }
        offset: 0
        limit: 3
        orderBy: { column: "id", order: "desc" }
      ) {
        id
        title
      }
    }
  `

  expect(await client.query({ query: query })).toMatchSnapshot()
})

test('test multiple queries in a request', async () => {
  const query = gql`
    query {
      queryPost(
        where: { id: { gt: "1" } }
        offset: 0
        limit: 3
        orderBy: { column: "id", order: "desc" }
      ) {
        id
        title
      },
      
      getAuthor(id: "1") {
        id
        email
      }
    }
  `

  expect(await client.query({ query: query })).toMatchSnapshot()
})
