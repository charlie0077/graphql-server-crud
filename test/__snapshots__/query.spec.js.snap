// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test aggregation 1`] = `
Object {
  "data": Object {
    "queryAuthorJoinPost": Array [
      Object {
        "__typename": "AuthorJoinPostWithAggregation",
        "age__avg": 88,
        "email": "vilurazu@vuse.tk",
        "id": "1",
      },
      Object {
        "__typename": "AuthorJoinPostWithAggregation",
        "age__avg": 32,
        "email": "pi@ga.co.uk",
        "id": "6",
      },
      Object {
        "__typename": "AuthorJoinPostWithAggregation",
        "age__avg": 54,
        "email": "on@puveca.in",
        "id": "8",
      },
      Object {
        "__typename": "AuthorJoinPostWithAggregation",
        "age__avg": 35,
        "email": "cor@viiru.jp",
        "id": "9",
      },
      Object {
        "__typename": "AuthorJoinPostWithAggregation",
        "age__avg": 33,
        "email": "ijsettav@mo.tl",
        "id": "10",
      },
      Object {
        "__typename": "AuthorJoinPostWithAggregation",
        "age__avg": 34,
        "email": "ige@pib.ph",
        "id": "11",
      },
      Object {
        "__typename": "AuthorJoinPostWithAggregation",
        "age__avg": 70,
        "email": "avo@ped.cc",
        "id": "12",
      },
      Object {
        "__typename": "AuthorJoinPostWithAggregation",
        "age__avg": 61,
        "email": "de@bosdufzi.ps",
        "id": "13",
      },
      Object {
        "__typename": "AuthorJoinPostWithAggregation",
        "age__avg": 69,
        "email": "ru@etu.er",
        "id": "14",
      },
      Object {
        "__typename": "AuthorJoinPostWithAggregation",
        "age__avg": 89,
        "email": "kezhupur@topevgos.gov",
        "id": "15",
      },
    ],
  },
  "loading": false,
  "networkStatus": 7,
  "stale": false,
}
`;

exports[`test complicate search 1`] = `
Object {
  "data": Object {
    "queryAuthor": Array [],
  },
  "loading": false,
  "networkStatus": 7,
  "stale": false,
}
`;

exports[`test limit and offset 1`] = `
Object {
  "data": Object {
    "queryPost": Array [
      Object {
        "__typename": "PostWithAggregation",
        "id": "3",
        "title": "Gokgor kezli lazgoh gaw nite mewseun do supma.",
      },
      Object {
        "__typename": "PostWithAggregation",
        "id": "4",
        "title": "Totunusa kazaha ruguk di zuvon mojfogow datiha ewa.",
      },
      Object {
        "__typename": "PostWithAggregation",
        "id": "6",
        "title": "Ewa oku lufbu erehec jeda ci ej ru.",
      },
    ],
  },
  "loading": false,
  "networkStatus": 7,
  "stale": false,
}
`;

exports[`test multiple queries in a request 1`] = `
Object {
  "data": Object {
    "getAuthor": Object {
      "__typename": "Author",
      "email": "vilurazu@vuse.tk",
      "id": "1",
    },
    "queryPost": Array [
      Object {
        "__typename": "PostWithAggregation",
        "id": "3333",
        "title": "new post 3333",
      },
      Object {
        "__typename": "PostWithAggregation",
        "id": "99",
        "title": "Nokcar conuk mi rub rokdul en tunnegov zojan.",
      },
      Object {
        "__typename": "PostWithAggregation",
        "id": "98",
        "title": "Jeaka govut iva liczizig denbu ecjob fov amofi.",
      },
    ],
  },
  "loading": false,
  "networkStatus": 7,
  "stale": false,
}
`;

exports[`test nested query 1`] = `
Object {
  "data": Object {
    "queryPost": Array [
      Object {
        "__typename": "PostWithAggregation",
        "author": Object {
          "__typename": "Author",
          "company": Object {
            "__typename": "Company",
            "domain": "nouco.cn",
          },
          "email": "ur@moej.wf",
        },
        "id": "2",
        "title": "new post 2222",
      },
      Object {
        "__typename": "PostWithAggregation",
        "author": Object {
          "__typename": "Author",
          "company": Object {
            "__typename": "Company",
            "domain": "nitdopfiz.ro",
          },
          "email": "vilurazu@vuse.tk",
        },
        "id": "3",
        "title": "Gokgor kezli lazgoh gaw nite mewseun do supma.",
      },
      Object {
        "__typename": "PostWithAggregation",
        "author": Object {
          "__typename": "Author",
          "company": Object {
            "__typename": "Company",
            "domain": "nouco.cn",
          },
          "email": "ur@moej.wf",
        },
        "id": "4",
        "title": "Totunusa kazaha ruguk di zuvon mojfogow datiha ewa.",
      },
    ],
  },
  "loading": false,
  "networkStatus": 7,
  "stale": false,
}
`;

exports[`test simple query 1`] = `
Object {
  "data": Object {
    "getAuthor": Object {
      "__typename": "Author",
      "email": "vilurazu@vuse.tk",
      "id": "1",
    },
  },
  "loading": false,
  "networkStatus": 7,
  "stale": false,
}
`;

exports[`test sort 1`] = `
Object {
  "data": Object {
    "queryPost": Array [
      Object {
        "__typename": "PostWithAggregation",
        "id": "3333",
        "title": "new post 3333",
      },
      Object {
        "__typename": "PostWithAggregation",
        "id": "99",
        "title": "Nokcar conuk mi rub rokdul en tunnegov zojan.",
      },
      Object {
        "__typename": "PostWithAggregation",
        "id": "98",
        "title": "Jeaka govut iva liczizig denbu ecjob fov amofi.",
      },
    ],
  },
  "loading": false,
  "networkStatus": 7,
  "stale": false,
}
`;
